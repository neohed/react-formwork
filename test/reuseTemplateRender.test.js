import React from 'react';
import ReuseTemplate from './forms/reuseTemplateForm';
import renderer from 'react-test-renderer';

test('Render a form with a reused template', () => {
    const component = renderer.create(
        <ReuseTemplate />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    /*
    // manually trigger the callback
    tree.props.onMouseEnter();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    */
});
