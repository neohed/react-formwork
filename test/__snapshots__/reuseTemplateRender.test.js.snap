// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a form with a reused template 1`] = `
<div>
  <form
    name="form01"
  >
    <div
      style={
        Object {
          "border": "2px solid red",
        }
      }
    >
      <p>
        Given Name:
      </p>
      <p>
        name
      </p>
      <p>
        <input
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="David"
        />
      </p>
    </div>
    <div
      style={
        Object {
          "border": "2px solid red",
        }
      }
    >
      <p>
        Email
      </p>
      <p>
        email
      </p>
      <p>
        <input
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="a@b.c"
        />
      </p>
    </div>
    <button
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
